File: EasyTravelAngular/src/app/models/blaBlaCar/trip.ts
import { Links } from './links';
import { Place } from './place';
import { Price } from './price';
import { Duration } from './duration';
import { Distance } from './distance';
import { Car } from './car';

export class Trip {
    links: Links;
    departure_date: string;
    arrival_date: string;
    departure_place: Place;
    arrival_place: Place;
    price: Price;
    seats_left: number;
    seats: number;
    duration: Duration;
    distance: Distance;
    car: Car;
    locations_to_display: string[];
}

File: EasyTravel.Core/Models/BlaBlaCar/Trip.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;
using EasyTravel.Contracts.Interfaces.Core;
using EasyTravel.Core.Models.Monitoring;
using Newtonsoft.Json;

namespace EasyTravel.Core.Models.BlaBlaCar
{
    public class Trip : ITrip
    {
        public int Id { get; set; }
        [JsonProperty("links")]
        public Links Links { get; set; }
        [JsonProperty("departure_date")]
        public DateTime DepartureDate { get; set; }
        [JsonProperty("arrival_date")]
        public DateTime ArrivalDate { get; set; }
        [JsonProperty("departure_place")]
        public Place DeparturePlace { get; set; }
        [JsonProperty("arrival_place")]
        public Place ArrivalPlace { get; set; }
        [JsonProperty("price")]
        public Price Price { get; set; }
        [JsonProperty("seats_left")]
        public int SeatsLeft { get; set; }
        [JsonProperty("seats")]
        public int Seats { get; set; }
        [JsonProperty("duration")]
        public Duration Duration { get; set; }
        [JsonProperty("distance")]
        public Distance Distance { get; set; }
        [JsonProperty("car")]
        public Car Car { get; set; }
        [NotMapped]
        [JsonProperty("locations_to_display")]
        public string[] LocationsToDisplay { get; set; }
    }
}
